de.dlr.sc.virsat.model.extension.fdir.ui/src/de/dlr/sc/virsat/model/extension/fdir/ui/diagram/ft/FaultTreeDiagramToolBehaviorProvider.java

```diff
@@ -118,4 +128,66 @@ public Object getToolTip(GraphicsAlgorithm ga) {
 		
 		return super.getToolTip(ga);
 	}
+	
+	@Override
+	public IPaletteCompartmentEntry[] getPalette() {
+		final String SIGNATURE = "getPaletteCompartments()";
+		boolean info = T.racer().info();
+		if (info) {
+			T.racer().entering(DefaultToolBehaviorProvider.class, SIGNATURE, new Object[0]);

```

Comment:
What exactly is this? :D
I dont remember seeing this anywhere else in our diagram code.

Reply:
Its internal stuff used in the overwritten method that I chose to keep since I dont know exactly what it does either. :D Not sure if omitting it causes undesirable behaviour.